tests/tests/htlc_tests.cpp

```diff
@@ -177,6 +177,448 @@ try {
+      update_operation.fee = db.current_fee_schedule().calculate_fee( update_operation );
+      trx.operations.push_back( update_operation );
+      sign(trx, joker_private_key);
+      GRAPHENE_CHECK_THROW( PUSH_TX( db, trx, ~0 ), fc::exception );
+      trx.clear();      
+   }
+
+   // joker sends a redeem operation to claim the funds for bob
+   {
+      graphene::chain::htlc_redeem_operation update_operation;
+      update_operation.redeemer = joker_id;
+      update_operation.htlc_id = alice_htlc_id;
+      update_operation.preimage = pre_image;
+      update_operation.fee = db.current_fee_schedule().calculate_fee( update_operation );
+      trx.operations.push_back( update_operation );
+      sign(trx, joker_private_key);
+      PUSH_TX( db, trx, ~0 );
+      generate_block();
+      trx.clear();
+   }
+   // verify funds end up in Bob's account (3)
+   BOOST_CHECK_EQUAL( get_balance(bob_id,   graphene::chain::asset_id_type()), bobs_balance + 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+   // verify funds remain out of Alice's acount ( 100 - 3 - transaction fee )
+   BOOST_CHECK_EQUAL( get_balance(alice_id, graphene::chain::asset_id_type()), 9057813 );
+   // verify all three get notified
+   std::vector<operation_history_object> history = get_operation_history(alice_id);
+   BOOST_CHECK_EQUAL( history.size(), alice_num_history + 1);
+   display_operation_history{ history, alice };
+   history = get_operation_history(bob_id);
+   BOOST_CHECK_EQUAL( history.size(), bob_num_history + 1);
+   display_operation_history{ history, bob };
+   history = get_operation_history(joker_id);
+   BOOST_CHECK_EQUAL( history.size(), joker_num_history + 1);
+   display_operation_history{ history, joker };
+
+   // Alice attempts to put a proposal out with a preimage size of 0 (which is valid)
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      graphene::chain::proposal_create_operation prop1;
+      prop1.fee_paying_account = alice_id;
+      prop1.expiration_time = db.head_block_time() + 100;
+      prop1.proposed_ops.emplace_back(create_operation);
+      trx.operations.push_back(prop1);
+      sign(trx, alice_private_key);
+      PUSH_TX(db, trx, ~0);
+      trx.clear();
+   }
+   // Alice attempts to put a proposal out with a memo (which is valid)
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
```

Comment:
Alice no longer has 100 coins. Also this test case is not about direct transfers.